roles/openshift_metrics/tasks/generate_certificates.yaml

```diff
@@ -36,14 +36,186 @@
     get 'secret/{{ sa_secret.stdout }}'
     --template '{{ '{{index .data "ca.crt"}}' }}'
   register: sa_secret
+- name: read files for the heapster secret
+  command: base64 --wrap 0 "{{ mktemp.stdout }}/certs/heapster.{{ file }}"
+  register: heapster_secret
+  with_items:
+  - cert
+  - key
+  loop_control:
+    loop_var: file
 - name: generate heapster secret template
   template:
-    src: heapster_secrets.j2
+    src: secret.j2
     dest: "{{ mktemp.stdout }}/templates/heapster_secrets.yaml"
   vars:
-    cert: >
-      {{ lookup('file', '{{ mktemp.stdout }}/certs/heapster.cert')|b64encode }}
-    key: >
-      {{ lookup('file', '{{ mktemp.stdout }}/certs/heapster.key')|b64encode }}
-    client_ca: "{{ sa_secret.stdout }}"
-    allowed_users: "{{ 'system:master-proxy'|b64encode }}"
+    name: heapster-secrets
+    labels:
+      metrics-infra: heapster
+    data:
+      heapster.cert: "{{ heapster_secret.results[0].stdout }}"
+      heapster.key: "{{ heapster_secret.results[1].stdout }}"
+      heapster.client-ca: "{{ sa_secret.stdout }}"
+      heapster.allowed-users: "{{ 'system:master-proxy'|b64encode }}"
+- name: generate hawkular-metrics certificates
+  include: setup_certificate.yaml
+  vars:
+    component: hawkular-metrics
+    hostnames: "hawkular-metrics,{{ hawkular_metrics_hostname }}"
+- name: generate hawkular-cassandra certificates
+  include: setup_certificate.yaml
+  vars:
+    component: hawkular-cassandra
+    hostnames: hawkular-cassandra
+# TODO keytool as dependency?  move key/trust store generation to containers?
```

Comment:
we are still having this debate for logging. currently we are using keytool within Ansible, I'm hoping to get some further feedback from the Ansible team regarding their stance on this.